Issue while parsing parcelable ParseUser object

This Stacktrace appeared while passing and parsing ParseUser object between activities.
This is not occurred repeatedly but through my investigation it occur when there is something present in operation set queue of ParseUser object. 
We can also check in stacktrace that reason for that might be `com.parse.ParseFieldOperations$5.decode(ParseFieldOperations.java:126)`
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{in.app/in.app.UPActivity}: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
                                                                    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2308)
                                                                    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2368)
                                                                    at android.app.ActivityThread.access$800(ActivityThread.java:144)
                                                                    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1285)
                                                                    at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                    at android.os.Looper.loop(Looper.java:135)
                                                                    at android.app.ActivityThread.main(ActivityThread.java:5233)
                                                                    at java.lang.reflect.Method.invoke(Native Method)
                                                                    at java.lang.reflect.Method.invoke(Method.java:372)
                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898)
                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
                                                                    at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:102)
                                                                 Caused by: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
                                                                    at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
                                                                    at java.util.ArrayList.set(ArrayList.java:481)
                                                                    at com.parse.ParseFieldOperations$5.decode(ParseFieldOperations.java:126)
                                                                    at com.parse.ParseFieldOperations.decode(ParseFieldOperations.java:235)
                                                                    at com.parse.ParseParcelDecoder.decode(ParseParcelDecoder.java:60)
                                                                    at com.parse.ParseOperationSet.fromParcel(ParseOperationSet.java:164)
                                                                    at com.parse.ParseObject.createFromParcel(ParseObject.java:4329)
                                                                    at com.parse.ParseObject$52.createFromParcel(ParseObject.java:4304)
                                                                    at com.parse.ParseObject$52.createFromParcel(ParseObject.java:4301)
                                                                    at android.os.Parcel.readParcelable(Parcel.java:2246)
                                                                    at android.os.Parcel.readValue(Parcel.java:2146)
                                                                    at android.os.Parcel.readArrayMapInternal(Parcel.java:2479)
                                                                    at android.os.BaseBundle.unparcel(BaseBundle.java:221)
                                                                    at android.os.Bundle.getParcelable(Bundle.java:804)
                                                                    at android.content.Intent.getParcelableExtra(Intent.java:4886)
                                                                    at in.app.UPActivity.onCreate(UPActivity.java:88)
                                                                    at android.app.Activity.performCreate(Activity.java:6001)
                                                                    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
                                                                    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2261)
                                                                    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2368) 
                                                                    at android.app.ActivityThread.access$800(ActivityThread.java:144) 
                                                                    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1285) 
                                                                    at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                    at android.os.Looper.loop(Looper.java:135) 
                                                                    at android.app.ActivityThread.main(ActivityThread.java:5233) 
                                                                    at java.lang.reflect.Method.invoke(Native Method) 
                                                                    at java.lang.reflect.Method.invoke(Method.java:372) 
                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898) 
                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693) 
                                                                    at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:102)
```

--- a/Parse/src/main/java/com/parse/ParseFieldOperations.java
+++ b/Parse/src/main/java/com/parse/ParseFieldOperations.java
@@ -105,7 +105,7 @@ import java.util.Set;
         int size = source.readInt();
         List<Object> list = new ArrayList<>(size);
         for (int i = 0; i < size; i++) {
-          list.set(i, decoder.decode(source));
+          list.add(i, decoder.decode(source));
         }
         return new ParseAddOperation(list);
       }
@@ -123,7 +123,7 @@ import java.util.Set;
         int size = source.readInt();
         List<Object> list = new ArrayList<>(size);
         for (int i = 0; i < size; i++) {
-          list.set(i, decoder.decode(source));
+          list.add(i, decoder.decode(source));
         }
         return new ParseAddUniqueOperation(list);
       }
@@ -141,7 +141,7 @@ import java.util.Set;
         int size = source.readInt();
         List<Object> list = new ArrayList<>(size);
         for (int i = 0; i < size; i++) {
-          list.set(i, decoder.decode(source));
+          list.add(i, decoder.decode(source));
         }
         return new ParseRemoveOperation(list);
       }

